tsml_eval/sktime_estimators/classification/deep_learning/inception_time.py

```diff
@@ -80,31 +78,49 @@ def __init__(
         self.random_state = random_state
         self.verbose = verbose
         self.classifers_ = []
+
+        super(InceptionTimeClassifier, self).__init__()
+
+    def _fit(self, X, y):
+        self.classifers_ = []
+        rng = check_random_state(self.random_state)
+
         for _ in range(0, self.n_classifiers):
             cls = IndividualInceptionTimeClassifier(
-                n_filters=n_filters,
-                use_bottleneck=use_bottleneck,
-                bottleneck_size=bottleneck_size,
-                depth=depth,
-                kernel_size=kernel_size,
-                batch_size=batch_size,
-                nb_epochs=nb_epochs,
-                callbacks=callbacks,
-                random_state=random_state,
-                verbose=verbose,
+                n_filters=self.n_filters,
+                use_bottleneck=self.use_bottleneck,
+                bottleneck_size=self.bottleneck_size,
+                depth=self.depth,
+                kernel_size=self.kernel_size,
+                batch_size=self.batch_size,
+                nb_epochs=self.nb_epochs,
+                callbacks=self.callbacks,
+                random_state=rng.randint(0, np.iinfo(np.int32).max),
+                verbose=self.verbose,
             )
+            cls.fit(X, y)
             self.classifers_.append(cls)
 
-    def _fit(self, X, y):
-        for i in range(0, self.n_classifiers):
-            self.classifiers_[i].fit(X, y)
         return self
 
     def _predict(self, X) -> np.ndarray:
-        return None
+        rng = check_random_state(self.random_state)
+        return np.array(
+            [
+                self.classes_[int(rng.choice(np.flatnonzero(prob == prob.max())))]
+                for prob in self._predict_proba(X)
+            ]
+        )
 
     def _predict_proba(self, X) -> np.ndarray:
-        return None
+        probs = np.zeros((X.shape[0], self.n_classes_))
+
+        for cls in self.classifers_:
+            probs += cls.predict_proba(X)
```

Comment:
I think you need to do this (from James's implementation) in predict_proba
        # check if binary classification
        if probs.shape[1] == 1:
        # first column is probability of class 0 and second is of class 1
            probs = np.hstack([1 - probs, probs])